New ports added to the code

costinm opened the pull request:
Update the doc for ports in use to reflect reality.

The HBONE port is not implemented yet, but we are starting work, better to reserve it.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

istio-testing commented:
@costinm: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint_istio.io | 086fd8ad2a64490eb48ee6042348bbe5d5dad314 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/9910/lint_istio.io/1402669109912539136) | `/test lint_istio.io`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

ericvn commented on content/en/docs/ops/deployment/requirements/index.md:
```diff
@@ -94,11 +94,15 @@ The following ports and protocols are used by the Istio sidecar proxy (Envoy).
 | 15020 | HTTP | Merged Prometheus telemetry from Istio agent, Envoy, and application | No |
 | 15021 | HTTP | Health checks | No |
 | 15090 | HTTP | Envoy Prometheus telemetry | No |
+| 15053 | DNS  | DNS port, if capture is enabled | Yes |
```
If keeping the ports in order, this port is not.

ericvn commented on content/en/docs/ops/deployment/requirements/index.md:
```diff
@@ -94,11 +94,15 @@ The following ports and protocols are used by the Istio sidecar proxy (Envoy).
 | 15021 | HTTP | Health checks | No |
 | 15090 | HTTP | Envoy Prometheus telemetry | No |
+| 15053 | DNS  | DNS port, if capture is enabled | Yes |
+
```
The linter will also complain about the extra blank line being added.

istio-testing commented:
@costinm: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint_istio.io | 243fcfd3be96c6507e1e610cb2d8683e674a5516 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/9910/lint_istio.io/1403125353907490816) | `/test lint_istio.io`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

costinm commented:
Fixed, also reordered the 443/8080 if we want ports ordered.

Updated the text a bit for the plaintext ports to make it a bit more clear they are intended for 'secure networks',
and added the container port for webhooks which was missing.

kyessenov commented on content/en/docs/ops/deployment/requirements/index.md:
```diff
@@ -94,21 +94,25 @@ The following ports and protocols are used by the Istio sidecar proxy (Envoy).
 | 15006 | TCP | Envoy inbound | No |
-| 15008 | TCP | Envoy tunnel port (inbound) | No |
+| 15008 | H2 | HBONE mTLS tunnel port | No |
+| 15009 | H2C | HBONE port for secure networks | No |
```
This is not ready, right? Why are we publishing it already.

costinm replied:
15008 was already published - I just updated the doc.

And we are starting to use them ( via bootstrap or envoy filter ) - should be reserved and documented.
